core/src/main/java/org/elasticsearch/indices/store/TransportNodesListShardStoreMetaData.java

```diff
@@ -150,7 +150,10 @@ private StoreFilesMetaData listStoreMetaData(ShardId shardId) throws IOException
             if (shardPath == null) {
                 return new StoreFilesMetaData(shardId, Store.MetadataSnapshot.EMPTY);
             }
-            return new StoreFilesMetaData(shardId, Store.readMetadataSnapshot(shardPath.resolveIndex(), shardId, logger));
+            // note that this may fail if it can't get access to the shard lock. Since we check above there is an active shard, this means:
+            // 1) a shard is being constructed, which means the master will not use a copy of this replcia
```

Comment:
hehe, yeah - ADD or the works :)
2) A shard is shutting down and has not cleared it's content within lock timeout. In this case the master may not reuse local resources.